sound/soc/codecs/rt1308-sdw.c

```diff
@@ -132,9 +132,9 @@ static int rt1308_read_prop(struct sdw_slave *slave)
 	/* for sink */
 	nval = hweight32(prop->sink_ports);
 	num_of_ports += nval;
-	prop->sink_dpn_prop = devm_kcalloc(&slave->dev, nval,
-						sizeof(*prop->sink_dpn_prop),
-						GFP_KERNEL);
+	prop->sink_dpn_prop = devm_kzalloc(&slave->dev,
+					   nval * sizeof(*prop->sink_dpn_prop),
+					   GFP_KERNEL);
```

Comment:
`kcalloc()` just like `calloc()` initialises memory to 0, this patch seems redundant to me.

Reply:
without this change the kernel bombs out with an uninitialized variable in that structure.

Reply:
never mind, one of those Friday afternoon tests ...